projects/ngrx.io/content/guide/component-store/usage.md

```diff
@@ -0,0 +1,240 @@
+* **Local UI State**. The state within a component itself. E.g. `isEnabled` toggle state of Toggle Component.
+
+<figure>
+  <img src="generated/images/guide/component-store/types-of-state.png" alt="Types of state in a typical SPA" width="100%" height="100%" />
+</figure>
+
+There are more types of states, but these are the most important ones in the context of state management.
+
+<div class="alert is-important">
+
+Synchronizing these states is one of the most complex tasks that developers have to solve.
+
+</div>
+
+Here is a small example to demonstrate how even a simple task might involve all of them:
+
+1. The user opens the page at specific URL - "https://TheBestMoviesOfAllTimeEver.com/favorites" - that changes ***URL State***. 
+1. URL had a path for a specific tab - "favorites" - that selection becomes part of ***Client State***. 
+1. Now, that results in API calls to the backend to get the data of the movies that user marked as "favorites" - we get the snapshot of ***Persisted State***.
+1. The Toggle Component that lives next to *"is favorite"* label is turned ON. The "ON" state is derived from the data that application received and passed to Toggle Component through `@Input() isEnabled: boolean`. The component itself is not aware of *Persisted State*, or what it even means to be ON in the context of the rest of the application. All it knows that it needs to be visually displayed as ON - `isEnabled` state is ***Local UI State***.
+1. The user might decided that this movie is no longer their favorite, and would click the Toggle button. The *Local UI State* of the component is then changed, the `@Output() changed` event is emitted and picked up by the container component, which then would call the backend to update the new *Persisted State*. 
+
+This was a very simple example, typically developers have to solve many more interconnected problems. What if the user is not logged in? Should we wait until the new favorite state is persisted to the backend and only then show disabled state, or should we do this optimistically? and etc.
+
+<div class="alert is-helpful">
+
+The understanding of these state would help us define the usage of ComponentStore.
+
+</div>
+
+## Use Case 1: Local UI State
+
+### Example 1: ComponentStore as part of component
+
+The simplest example of the usage of `ComponentStore` would be the **reactive *Local UI State***.
+
+One of the ways to improve the performance of the application is to use `OnPush` change detection strategy. However, contrary to the popular belief, we do not always need to tell Angular's change detection to `markForCheck()` or `detectChanges()` (or the Ivy's alternatives). As pointed in [the article on change detection](https://indepth.dev/the-last-guide-for-angular-change-detection-youll-ever-need/), if the event originates from the component it would be checked.
```

Comment:
```suggestion
One of the ways to improve the performance of the application is to use the `OnPush` change detection strategy. However, contrary to the popular belief, we do not always need to tell Angular's change detection to `markForCheck()` or `detectChanges()` (or the Angular Ivy alternatives). As pointed out in [this article on change detection](https://indepth.dev/the-last-guide-for-angular-change-detection-youll-ever-need/), if the event originates from the component itself, it will be dirty checked.
```